node/lib/util/commit.js

```diff
@@ -889,24 +889,47 @@ exports.getAmendStatus = co.wrap(function *(repo, options) {
     const head = yield repo.getHeadCommit();
     const headTree = yield head.getTree();
 
-    const newUrls =
-                 yield SubmoduleConfigUtil.getSubmodulesFromCommit(repo, head);
+    const getSubmodulesFromCommit = SubmoduleConfigUtil.getSubmodulesFromCommit;
+    const newUrls = yield getSubmodulesFromCommit(repo, head);
 
     // Read the state of the commits in the commit before the one to be
     // amended.
 
-    let oldUrls = {};
-    const parent = yield GitUtil.getParentCommit(repo, head);
-    let parentTree = null;
-    if (null !== parent) {
-        parentTree = yield parent.getTree();
-        oldUrls =
-               yield SubmoduleConfigUtil.getSubmodulesFromCommit(repo, parent);
-    }
-    const diff =
-               yield NodeGit.Diff.treeToTree(repo, parentTree, headTree, null);
-    const changes = yield SubmoduleUtil.getSubmoduleChangesFromDiff(diff,
-                                                                    true);
+    const oldUrls = {};
+    const parents = yield head.getParents();
+
+    const merged = {};
```

Comment:
Need more documentation about what your algorithm is here -- e.g., what are `merged` and `changes` for? I think it might be a good point to factor this logic into a new method, e.g., `computeAmendSubmoduleChanges`.

Reply:
I'd also recommend that instead of having a `merged` and `changes` approach like this, you return a single recording the correct new and old SHAs so you don't have a special case for merge later.

Reply:
I think I need a special case for merge.  For submodules changed in this commit, we do want to amend them. For ones not changed, we don't.

Reply:
But I think the algorithm for that is to amend submodule commits whose SHA doesn't exist in the tree of any immediate parent. The case of having one parent doesn't seem special.